### PR TITLE
Bring nupic requirement up to 0.2.12 as well as make py.test requirem…

### PR DESCRIPTION
**Pull request by oxtopus**
…ent consistent

When you're ready with a nupic@0.2.12 release, @rhyolight, @jcasner, and @scottpurdy

### CODE REVIEW

**Comment by vitaly-krugl** [comment]
Taurus Engine and Taurus Metric Collectors projects should have pytest in their requirements, too.

File: grok/requirements.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 dbutils==1.1
 numpy==1.9.2
 mysql-python==1.2.5
-pytest==2.4.2
+pytest==2.5.1
```
Comment (vitaly-krugl) [inline grok/requirements.txt]: Move to the "# required by tests" section?

**Comment by vitaly-krugl** [comment]
Reviewed and added some feedback about pytest.

**Comment by scottpurdy** [comment]
:+1: aside from @vitaly-krugl feedback

**Comment by jcasner** [comment]
Closing in favor of #298 which includes this commit as well as some other fixes that should all get merged together.

**Pull request closed without merging**